### PR TITLE
Comparison and Shifts with Literals

### PR DESCRIPTION
**Pull request by jzhang919**
Added Features:

+ UnsignedBVExpr now supports left and logical right shifts. Also supports unsigned comparisons with literals.
+ SignedBVExpr now supports left and arithmetic right shifts. Also supports signed comparisons with literals. 

+ Added tests in test_expr.py to confirm correctness of all added operators.

### CODE REVIEW

File: aigerbv/common.py [hunk 1]
```diff
@@ -11,447 +11,445 @@
 
 
 def _fresh():
-    return str(uuid1())
+	return str(uuid1())
```
Comment (mvcisback) [inline aigerbv/common.py]: Is there a reason to change the indentation? I would prefer to stick to 4 spaces since that's what I've used throughout.

Comment (mvcisback) [reply aigerbv/common.py]: Also, it makes the pull-request unreadable..

File: aigerbv/expr.py [hunk 2]
```diff
@@ -4,143 +4,156 @@
...
-        return _unary_gate(cmn.abs_gate, self)
+	def __abs__(self):
+		return _unary_gate(cmn.abs_gate, self)
 
 
 Expr = Union[UnsignedBVExpr, SignedBVExpr]
 
 
+def _shift_gate(gate, expr, n_bits):
+	wordlen = expr.size + n_bits if gate == cmn.left_shift_gate else expr.size
```
Comment (mvcisback) [inline aigerbv/expr.py]: Hmm, I guess there's an implicit question about semantics here.

I personally don't think left shift should add extra bits, but rather have the bits "fall off the end". This matches most implementations of word level shift that I have seen.

Comment (mvcisback) [reply aigerbv/expr.py]: Also, given the above comments, one could just have:

```python
from functools import partial

def _shift_gate(gate, expr, n_bits):
    return _unary_gate(gate=partial(gate, shift=n_bits), expr=expr)
```

File: aigerbv/expr.py [hunk 3]
```diff
@@ -4,143 +4,156 @@
 
 from aigerbv import aigbv
 from aigerbv import common as cmn
-
+import inspect
```
Comment (mvcisback) [inline aigerbv/expr.py]: Did you forget to commit that code that use this? presumably it's for testing literals?

**Review by mvcisback — CHANGES_REQUESTED** [review]
Overall, looks like a very good step in the right direction.

I made some comments in my code review for requested changes.

I apologize that the continuous integration seems to be broken at the moment. flake8 updated their API in a breaking way and I didn't pin the version......

File: aigerbv/common.py (hunk 1 above)
Comment (jzhang919) [reply aigerbv/common.py]: Will revert. Change was due to my auto style-checker script; will ensure that 4 spaces are used in the future. 👍

File: aigerbv/expr.py (hunk 2 above)
Comment (jzhang919) [reply aigerbv/expr.py]: Sounds good. Committed new version where left shifts will have bits "fall off the end," e.g
0b111 << 1 = 0b110.

EDIT: See test_sll in test_expr.py for new behavior. Bits now fall off; also, signed binary representation is one's complement.

File: aigerbv/expr.py (hunk 3 above)
Comment (jzhang919) [reply aigerbv/expr.py]: Will remove. I used inspect for checking class type in some other tests I created to better understand the aiger functions.

**Review by mvcisback — CHANGES_REQUESTED** [review]
Looks good! Just fix the flake8 issues so the continuous integration passes and I will merge.

**Review by mvcisback — APPROVED** [review]

**Pull request merged**